sktime/base.py

```diff
@@ -301,14 +301,20 @@ def __enter__(self):
             if isinstance(d, np.ndarray):
                 self.old_writable_flags.append(d.flags.writeable)
                 # set ndarray writabe flags to false
-                d.flags.writeable = False
+                try:
+                    d.setflags(write=False)
+                except:
+                    ...  # although this should not happen, occasionally it does due to arrays stemming from torch etc
```

Comment:
what is "it"

Reply:
should a warning be printed here or doesn't matter? what does it mean if the line in the try part doesn't get executed?

Reply:
"it" is an exception being raised when trying to set the flag. Setting the flag itself is intended to make it impossible that the `fit` (or `partial_fit`) mutate the input data, i.e. `estimator.fit(x)` should leave `x` untouched. When in this method `d` is a numpy array which stems from a torch tensor (or vice versa), then the data is coupled, i.e. changing one also changes the other. These kind of arrays are special in the sense that they don't support immutability. This is why I introduced the try-catch :)

Reply:
might be helpful for a future developer to comment this in?